crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -328,3 +412,63 @@ fn check_directives(directives: &[Positioned<Directive>]) -> ServerResult<()> {
     }
     Ok(())
 }
+/// Given a node, estimate the number of output nodes it will produce.
+fn estimate_output_nodes_for_curr_node(
+    f: &Positioned<Field>,
+    variables: &Variables,
+    limits: &Limits,
+) -> u64 {
+    let mut current_count = 1;
+
+    if check_is_connection(f) {
+        // Set defaults for connection field
+
+        // If the args 'first' or 'last' is set, then we should use that as the count
+        let first_arg = f.node.get_argument("first");
+        let last_arg = f.node.get_argument("last");
+
+        let first_count = extract_limit(first_arg, variables).unwrap_or(limits.default_page_size);
+        let last_count = extract_limit(last_arg, variables).unwrap_or(limits.default_page_size);
+
+        current_count = match (first_arg, last_arg) {
+            (Some(_), None) => first_count,
+            (None, Some(_)) => last_count,
+            (Some(_), Some(_)) => std::cmp::min(first_count, last_count),
+            (None, None) => limits.default_page_size,
+        };
+    }
+
+    current_count
```

Comment:
I think you can probably simplify the calculation of `current_count`, by ignoring the case where both first and last are supplied, because this will result in an error elsewhere, anyway (the simplification I'm proposing would result in treating `first` and `last` both being provided the same as just `first` being provided).

```suggestion
    if !is_connection(f) {
        1
    } else {
        // If the args 'first' or 'last' is set, then we should use that as the count
        let first_arg = f.node.get_argument("first");
        let last_arg = f.node.get_argument("last");

        extract_limit(first_arg, variables)
            .or_else(|| extract_limit(last_arg, variables))
            .unwrap_or(self.default_page_size).
    }
```

Reply:
Yeah that's fair enough. Gotta tell myself to let go, let go